src/__test__/detailsPage/detailsPage.test.js

```diff
@@ -0,0 +1,67 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import App from '../../App';
+
+const jsonCountry = {
+  json: () => Promise.resolve([
+    {
+      name: 'Rwanda',
+      population: '12952',
+      iso2: 'RW',
+    },
+  ]),
+};
+
+const jsonCity = {
+  json: () => Promise.resolve([
+    {
+      name: 'Kigali',
+      population: '745261',
+      is_capital: true,
+    },
+    {
+      name: 'Nyanza',
+      population: '225209',
+      is_capital: false,
+    },
+  ]),
+};
+
+describe('Details Page for Rwanda', () => {
+  it('should render the Rwanda details page', async () => {
+    window.fetch = jest.fn().mockImplementationOnce((url) => {
+      if (url.includes('city')) {
+        return jsonCity;
+      }
```

Comment:
[ Optional Changes ]

- [ ] The console is showing warnings on your test for this component relating to prop types. Please check on that 👍🏽